test/test_stone.py

```diff
@@ -1315,6 +1445,340 @@ def test_forward_reference(self):
         self.assertEqual(
             api.namespaces['test'].data_types[0].fields[0].default.tag_name, 'a')
 
+    def test_struct_patch_semantics(self):
```

Comment:
How about a test patching a struct that hasn't been defined?

Reply:
This exists already, look for comment `# Try patching type without pre-existing type`